--- a/src/SessionProvider.php	2026-07-27 10:43:45.434667756 +0000
+++ b/src/SessionProvider.php	2026-07-27 10:43:45.434667756 +0000
@@ -174,7 +174,7 @@
 			return $this->makeException( 'mwoauth-invalid-authorization-wrong-wiki', $wiki );
 		}
 
-		$username = Utils::getCentralUserNameFromId( $access->getUserId() );
+		$username = Utils::getCentralUserNameFromId( $access->getUserId(), 'raw' );
 		if ( $username === false || $username === '' ) {
 			return $this->makeException( 'mwoauth-invalid-authorization-invalid-user',
 				Message::rawParam( Utils::getErrorLink( 'E008' ) )
