--- a/src/Services/FormatterBuilder.php	2026-07-15 08:29:41.078360443 +0000
+++ b/src/Services/FormatterBuilder.php	2026-07-15 09:56:24.441137709 +0000
@@ -4,6 +4,7 @@
 
 namespace Wikibase\LocalMedia\Services;
 
+use ParserOptions;
 use RequestContext;
 use ValueFormatters\FormatterOptions;
 use ValueFormatters\ValueFormatter;
@@ -27,7 +28,7 @@
 
 		if ( $snakFormat->isPossibleFormat( SnakFormatter::FORMAT_HTML_VERBOSE, $format ) ) {
 			return new InlineImageFormatter(
-				RequestContext::getMain()->getOutput()->parserOptions(),
+				ParserOptions::newFromContext( RequestContext::getMain() ),
 				$this->thumbLimits,
 				$options->getOption( ValueFormatter::OPT_LANG ),
 				new LocalImageLinker(),
